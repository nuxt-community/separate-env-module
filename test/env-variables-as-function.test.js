jest.setTimeout(60000)

const { Nuxt, Builder } = require('nuxt-edge')
const { waitFor } = require('@nuxt/utils-edge')
const getPort = require('get-port')

describe('correctly assign env variables from function', () => {
  let nuxt

  beforeAll(async () => {
    nuxt = new Nuxt(require('./fixture/env-variables-as-function/nuxt.config'))
    await nuxt.ready()
    await new Builder(nuxt).build()
    await waitFor(2000)
    await nuxt.listen(await getPort())
  })

  afterAll(async () => {
    await nuxt.close()
  })

  test('render', async () => {
    const { html } = await nuxt.renderRoute('/')
    expect(html).toMatchSnapshot()
  })
})
