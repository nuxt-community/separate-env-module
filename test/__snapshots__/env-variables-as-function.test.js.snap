// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly assign env variables from function render 1`] = `
"<!doctype html>
<html data-n-head-ssr >
  <head >
    <style data-vue-ssr-id=\\"68844ae4:0\\">
.nuxt-progress {
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  height: 2px;
  width: 0%;
  opacity: 1;
  transition: width 0.1s, opacity 0.4s;
  background-color: black;
  z-index: 999999;
}
.nuxt-progress.nuxt-progress-notransition {
  transition: none;
}
.nuxt-progress-failed {
  background-color: red;
}
</style>
  </head>
  <body >
    <div data-server-rendered=\\"true\\" id=\\"__nuxt\\"><!----><div id=\\"__layout\\"><div id=\\"result\\"><h1>server</h1> <pre>{&quot;ONLY_CLIENT&quot;:&quot;undefined&quot;,&quot;ONLY_SERVER&quot;:&quot;yup&quot;,&quot;DIFFERENT&quot;:&quot;server&quot;,&quot;TYPE&quot;:&quot;undefined&quot;,&quot;normal&quot;:&quot;Hi&quot;}</pre></div></div></div><script>window.__NUXT__=(function(a,b){return {layout:\\"default\\",data:[{type:a,env:{ONLY_CLIENT:b,ONLY_SERVER:\\"yup\\",DIFFERENT:a,TYPE:b,normal:\\"Hi\\"}}],error:null,serverRendered:true,logs:[]}}(\\"server\\",\\"undefined\\"));</script><script src=\\"/_nuxt/runtime.js\\" defer></script><script src=\\"/_nuxt/pages/index.js\\" defer></script><script src=\\"/_nuxt/commons.app.js\\" defer></script><script src=\\"/_nuxt/vendors.app.js\\" defer></script><script src=\\"/_nuxt/app.js\\" defer></script>
  </body>
</html>
"
`;
